models/Element/AbstractElement.php

```diff
@@ -732,4 +735,71 @@ protected function renewInheritedProperties(): void
         $inheritedProperties = $this->getDao()->getProperties(true);
         $this->setProperties(array_merge($inheritedProperties, $myProperties));
     }
+
+    protected function retryableFunction(
+        ?Closure $beforeRetryables = null, 
+        ?Closure $retryableFunc = null,
+        ?Closure $onCommit = null, 
+        ?Closure $onBeforeRetry = null, 
+        ?Closure $onFailure = null,
+        int $maxRetries = 5,
+    ): void {
+        try {
+            if ($beforeRetryables instanceof Closure) {
+                $beforeRetryables();
+            }
+
+            for ($retries = 0; $retries < $maxRetries; $retries++) {
+                $this->beginTransaction();
+
+                try {
+                    if ($retryableFunc instanceof Closure) {
+                        $retryableFunc();
+                    }
+                    $this->commit();
+                    if ($onCommit instanceof Closure) {
+                        $onCommit();
+                    }
+
+                    break; // transaction was successfully completed, so we cancel the loop here -> no restart required
+                } catch (Exception $e) {
```

Comment:
```suggestion
                } catch (\Throwable $e) {
```
@kingjia90 `RetryableException` implements `Throwable` due to that it's never caught with `Exception`.

Reply:
oh.. thank you for pointing out, weird, wondering if the former code actually work, i just cut&paste from there.
I guess just
```suggestion
                } catch (\Throwable $e) {
```
would be fine, as it includes the former

<img width="891" height="209" alt="image" src="https://github.com/user-attachments/assets/3593261a-d911-4acb-ae67-dc86b6b7fde6" />